enterprise/reporting/client/client.go

```diff
@@ -0,0 +1,147 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	obskit "github.com/rudderlabs/rudder-observability-kit/go/labels"
+	"github.com/rudderlabs/rudder-server/jsonrs"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/utils/httputil"
+)
+
+const (
+	StatRequestTotalBytes     = "reporting_client_http_request_total_bytes"
+	StatTotalDurationsSeconds = "reporting_client_http_total_durations_seconds"
+	StatRequestLatency        = "reporting_client_http_request_latency"
+	StatHttpRequest           = "reporting_client_http_request"
+)
+
+const (
+	PathMetrics      Path = "/metrics?version=v1"
+	PathRecordErrors Path = "/recordErrors"
+	PathAggregates   Path = "/aggregates"
+)
+
+type Path string
+
+// Client handles sending metrics to the reporting service
+type Client struct {
+	httpClient          *http.Client
+	reportingServiceURL string
+	region              string
+	label               string
+	stats               stats.Stats
+	log                 logger.Logger
+	instanceID          string
+	path                Path
+}
+
+// New creates a new reporting client
+func New(reportingServiceURL string, path Path, conf *config.Config, log logger.Logger, stats stats.Stats) *Client {
+	reportingServiceURL = strings.TrimSuffix(reportingServiceURL, "/")
+	tr := &http.Transport{}
```

Comment:
why don't we use the default transport, or even provide options for configuration overrides?

Reply:
Good point, I started this client by copying on of the reporting.Send implementations.

Now that I see this again, it doesn't make sense to do it. I'will remove it